packages/core/src/utilities/VoxelManager.ts

```diff
@@ -283,25 +608,159 @@ export default class VoxelManager<T> {
+   * @param dimensions - The dimensions of the image volume.
+   * @param imageIds - The array of image IDs.
+   * @returns A VoxelManager instance for the image volume.
+   */
+  public static createImageVolumeVoxelManager({
+    dimensions,
+    imageIds,
+  }: {
+    dimensions: Point3;
+    imageIds: string[];
+  }): VoxelManager<number> {
+    const pixelsPerSlice = dimensions[0] * dimensions[1];
+
+    function getPixelInfo(index) {
+      const sliceIndex = Math.floor(index / pixelsPerSlice);
+      const imageId = imageIds[sliceIndex];
+      const image = cache.getImage(imageId);
+
+      if (!image) {
+        // Todo: better handle this case
+        console.warn(`Image not found for imageId: ${imageId}`);
+        return { pixelData: null, pixelIndex: null };
+      }
+
+      const pixelData = image.voxelManager.getScalarData();
+      const pixelIndex = index % pixelsPerSlice;
+
+      return { pixelData, pixelIndex };
+    }
+
+    function getVoxelValue(index) {
+      const { pixelData, pixelIndex } = getPixelInfo(index);
+
+      if (!pixelData || !pixelIndex) {
+        return null;
+      }
+
+      return pixelData[pixelIndex];
+    }
+
+    function setVoxelValue(index, v) {
+      const { pixelData, pixelIndex } = getPixelInfo(index);
+
+      if (pixelData[pixelIndex] === v) {
+        return false;
+      }
+
+      pixelData[pixelIndex] = v;
+      return true;
+    }
+
+    const voxelManager = new VoxelManager(
+      dimensions,
+      (index) => getVoxelValue(index),
+      (index, v) => setVoxelValue(index, v)
+    );
+
+    // @ts-ignore
+    voxelManager._getConstructor = () => {
+      const { pixelData } = getPixelInfo(0);
+      return pixelData?.constructor;
+    };
+
+    voxelManager.getMiddleSliceData = () => {
+      const middleSliceIndex = Math.floor(dimensions[2] / 2);
+      return voxelManager.getSliceData({
+        sliceIndex: middleSliceIndex,
+        slicePlane: 2,
+      });
+    };
+
+    // Todo: need a way to make it understand dirty status if pixel data is changed
+    voxelManager.getRange = () => {
+      // get all the pixel data
+      let minValue, maxValue;
+      for (const imageId of imageIds) {
+        const image = cache.getImage(imageId);
+
+        // min and max pixel value is correct, //todo this is not true
+        // for dynamically changing data such as labelmaps in segmentation
+        if (image.minPixelValue < minValue) {
+          minValue = image.minPixelValue;
+        }
+        if (image.maxPixelValue > maxValue) {
+          maxValue = image.maxPixelValue;
+        }
+      }
+      return [minValue, maxValue];
+    };
+
+    voxelManager._getScalarDataLength = () => {
+      const { pixelData } = getPixelInfo(0);
+      return pixelData.length * dimensions[2];
+    };
+
+    /**
+     * Retrieves the scalar data in a memory-inefficient manner.
+     * Useful for debugging, testing, or methods requiring all scalar data at once.
```

Comment:
It is important to add a comment that this is READ ONLY - that is, changes to the buffer are never reflected in the underlying data unless individually merged back.